src/market/market_state.py

```diff
@@ -0,0 +1,130 @@
+import math
+from typing import NamedTuple
+
+from solana.publickey import PublicKey
+from solana.rpc.api import Client
+
+from src._layouts.market import MINT_LAYOUT
+from src.utils import load_bytes_data
+
+
+class AccountFlags(NamedTuple):
+    initialized: bool = False
+    market: bool = False
+    open_orders: bool = False
+    request_queue: bool = False
+    event_queue: bool = False
+    bids: bool = False
+    asks: bool = False
+
+
+class MarketState(NamedTuple):
+    account_flags: AccountFlags = AccountFlags()
+    own_address: PublicKey = PublicKey(0)
+    vault_signer_nonce: int = 0
+    base_mint: PublicKey = PublicKey(0)
+    quote_mint: PublicKey = PublicKey(0)
+    base_vault: PublicKey = PublicKey(0)
+    base_deposits_total: int = 0
+    base_fees_accrued: int = 0
+    quote_vault: PublicKey = PublicKey(0)
+    quote_deposits_total: int = 0
+    quote_fees_accrued: int = 0
+    quote_dust_threshold: int = 0
+    request_queue: PublicKey = PublicKey(0)
+    event_queue: PublicKey = PublicKey(0)
+    bids: PublicKey = PublicKey(0)
+    asks: PublicKey = PublicKey(0)
+    base_lot_size: int = 1
+    quote_lot_size: int = 1
+    fee_rate_bps: int = 0
+    base_spl_token_decimals: int = 0
+    quote_spl_token_decimals: int = 0
+    program_id: PublicKey = PublicKey(0)
+    skip_preflight: bool = False
+    confirmations: int = 0
+
+    def base_spl_token_multiplier(self) -> int:
+        return 10 ** self.base_spl_token_decimals
+
+    def quote_spl_token_multiplier(self) -> int:
+        return 10 ** self.quote_spl_token_decimals
+
+    def base_spl_size_to_number(self, size: int) -> float:
+        return size / self.base_spl_token_multiplier()
+
+    def quote_spl_size_to_number(self, size: int) -> float:
+        return size / self.quote_spl_token_multiplier()
+
+    def price_lots_to_number(self, price: int) -> float:
+        return float(price * self.quote_lot_size * self.base_spl_token_multiplier()) / (
+            self.base_lot_size * self.quote_spl_token_multiplier()
+        )
+
+    def price_number_to_lots(self, price: float) -> int:
+        return int(
+            round(
+                (price * 10 ** self.quote_spl_token_multiplier() * self.base_lot_size)
+                / (10 ** self.base_spl_token_multiplier() * self.quote_lot_size)
+            )
+        )
+
+    def base_size_lots_to_number(self, size: int) -> float:
+        return float(size * self.base_lot_size) / self.base_spl_token_multiplier()
+
+    def base_size_number_to_lots(self, size: float) -> int:
+        return int(math.floor(size * 10 ** self.base_spl_token_decimals) / self.base_lot_size)
+
+
+def get_mint_decimals(conn: Client, mint_pub_key: PublicKey) -> int:
+    """Get the mint decimals from given public key."""
+    bytes_data = load_bytes_data(mint_pub_key, conn)
+    return MINT_LAYOUT.parse(bytes_data).decimals
+
+
+def create_account_flags(con) -> AccountFlags:
```

Comment:
Can this be an `__init__` method that takes an instance of a construct container? And add comment about `con: Any` is a contruct container.

Reply:
discussed over Discord, agree that this is find. Added comments about `construct` argument.